### PR TITLE
handle sharing of auras with people crossrealm in same group

### PR DESCRIPTION
**Pull request by mrbuds**
When sending addonmessage trough the "WHISPER" channel, first check if unit is on same realm
If not then check if on same group or raid and send the message trough the "PARTY" or "RAID" channel with a prefix "§§destinationName-destinationServer:message"

Then when receiving messages, when it is from channel "PARTY" or "RAID", check for the prefix, and check if your are the right recipient.

### CODE REVIEW

File: WeakAuras/Transmission.lua [hunk 1]
```diff
@@ -1675,6 +1675,18 @@ function WeakAuras.ImportString(str)
   end
 end
 
+function crossRealmSendCommMessage(prefix, text, chattype, target, queueName, callbackFn, callbackArg)
+  if chattype == "WHISPER" and target and not UnitIsSameServer(target) then
```
Comment (InfusOnWoW) [inline WeakAuras/Transmission.lua]: What does UnitIsSameServer do for merged realms?

File: WeakAuras/Transmission.lua [hunk 2]
```diff
@@ -1675,6 +1675,18 @@ function WeakAuras.ImportString(str)
...
+    if UnitInRaid(target) then
+      chattype = "RAID"
+    elseif UnitInParty(target) then
+      chattype = "PARTY"
+    end
```
Comment (InfusOnWoW) [inline WeakAuras/Transmission.lua]: Shouldn't that do:
else
 return

?

File: WeakAuras/Transmission.lua [hunk 3]
```diff
@@ -1675,6 +1675,18 @@ function WeakAuras.ImportString(str)
   end
 end
 
+function crossRealmSendCommMessage(prefix, text, chattype, target, queueName, callbackFn, callbackArg)
```
Comment (InfusOnWoW) [inline WeakAuras/Transmission.lua]: There's no point in having a chattype argument in here, as that is always "WHISPER".

File: WeakAuras/Transmission.lua (hunk 1 above)
Comment (mrbuds) [reply WeakAuras/Transmission.lua]: i trusted https://wow.gamepedia.com/API_UnitIsSameServer on this, and i just did my own tests, it returns true when unit is on a linked server

File: WeakAuras/Transmission.lua (hunk 2 above)
Comment (mrbuds) [reply WeakAuras/Transmission.lua]: yes

File: WeakAuras/Transmission.lua (hunk 3 above)
Comment (mrbuds) [reply WeakAuras/Transmission.lua]: indeed

File: WeakAuras/Transmission.lua (hunk 2 above)
Comment (mrbuds) [reply WeakAuras/Transmission.lua]: If that happen the whole PR can be reverted

Comment (InfusOnWoW) [reply WeakAuras/Transmission.lua]: We might not immediately recognize that. If the code would do what I suggest, it would just magically start working.

**Pull request merged**